ckcp/openshift_dev_setup.sh

```diff
@@ -275,20 +267,12 @@ patches:
         value: $ckcp_route " >>"$ckcp_temp_dir/kustomization.yaml"
 
   echo -n "  - kcp $kcp_version: "
-  kubectl apply -k "$ckcp_temp_dir" >/dev/null
-  # Check if ckcp pod status is Ready
-  local error="false"
-  kubectl wait --for=condition=Ready -n "$ns" pod -l=app=kcp-in-a-pod --timeout=120s >/dev/null 2>&1 || error="true"
-  if [ "$error" == "true" ]; then
-    printf "Debug: print out pod state\n"
-    kubectl -n "$ns" describe pods
-
-    printf "Debug: print out pod logs\n"
-    kubectl -n "$ns" logs "$(kubectl -n "$ns" get pods -l=app=kcp-in-a-pod -o name)"
-
-    exit 1
+  # Deploy ckcp until all resources are successfully appied to OCP cluster 
+  if ! timeout 120s bash -c "while ! kubectl apply -k $ckcp_temp_dir >/dev/null 2>&1; do printf '.'; sleep 10; done"; then
+    exit_error "ckcp deployment failed (timeout)"
```

Comment:
Why do you think that it is here acceptable to  erase error messages?
I understand that there is a retry loop but I would like at least to see what resource creation failed and for what reason during the last iteration.
""ckcp deployment failed (timeout)" is not useful for troubleshooting

Reply:
@fgiloux when the retry loop fails, we execute the command again, do you agree?
```
 if ! timeout 120s bash -c "while ! kubectl apply -k $ckcp_temp_dir >/dev/null 2>&1; do printf '.'; sleep 10; done"; then
    kubectl apply -k $ckcp_temp_dir >/dev/null
 fi
```

Reply:
Yes that would be a good solution. You would still keep the exit_error if the command is not successful. A comment stating "final trial with error display this time" or something similar would also be welcome

Reply:
when timeout is triggered, maybe there is a short period of time between the last failure and the timeout. So I didn't use the logic that I mentioned `when the retry loop fails, we execute the command again`. Probably it will get different result from the latest failure. I modified the logic which just prints the error message of the last failure.